docs/scripts/util/crawlAllLinks.ts

```diff
@@ -0,0 +1,40 @@
+import puppeteer from 'puppeteer';
+import { runArrayPromiseInOrder } from '.';
+
+type Links = {
+  pageUrl: string;
+  links: { href: string; tagName: string; tagText: string }[];
+};
+type AllPagesPaths = Map<string, Links>;
+
+export async function crawlAllLinks(pages: string[]) {
+  const allPagesPaths: AllPagesPaths = new Map();
+  await runArrayPromiseInOrder(pages, checkSitemapPath);
+
+  return allPagesPaths;
+
+  async function checkSitemapPath(pageUrl: string, pageIdx: string) {
```

Comment:
Is there a reason to define this function inside the other? I'd prefer to see it outside the `crawlAllLinks` so it's clearer what `checkSitemapPath` depends on.

Reply:
yes, because we don't use `checkSitemapPath` anywhere else. Having it inside `crawlAllLinks` to avoid creating an unnecessary closure.

Reply:
Totally appreciate the sentiment around avoiding the unnecessary closure, but not sure if the optimization is worth sacrificing readability

Reply:
*Having* nested function creates more unnecessary closure if I understand correctly, right?

For example, in the inner function you are accessing `allPagesPaths` from the outer function using closure. Wondering if we could avoid closure and accessing outer variable via closure, and instead just pass `allPagesPaths` implicitly.

IMO it's also weird to see a function below a return statement 💦

Reply:
Oops, after reading @calebpollman's comment, I realize I was talking about different closure, sorry 🤦 

I still agree with folks that moving the inner function outside is cleaner, but open to discussion!

Reply:
It's a functional programming best practice to avoid nested function declarations unless they are absolutely necessary

Reply:
Done ca25b298a5bcd666d81d4abc5cc6676175a77e6b